Lib/tarfile.py

```diff
@@ -1969,11 +1969,13 @@ def add(self, name, arcname=None, recursive=True, *, filter=None):
                 self.addfile(tarinfo, f)
 
         elif tarinfo.isdir():
-            self.addfile(tarinfo)
+            if tarinfo.name != '.':
+                self.addfile(tarinfo)
             if recursive:
                 for f in sorted(os.listdir(name)):
-                    self.add(os.path.join(name, f), os.path.join(arcname, f),
-                            recursive, filter=filter)
+                    if f != '.':
```

Comment:
Is this necessary? I don't believe `os.listdir` will ever include `'.'` in its results.

Reply:
I have made the requested changes; please review again